Content.Shared/Implants/SharedImplanterSystem.cs

```diff
@@ -125,42 +131,94 @@ public void Draw(EntityUid implanter, EntityUid user, EntityUid target, Implante
                     permanentFound = implantComp.Permanent;
-                    continue;
-                }
 
-                _container.Remove(implant, implantContainer);
-                implantComp.ImplantedEntity = null;
-                _container.Insert(implant, implanterContainer);
-                permanentFound = implantComp.Permanent;
-
-                var ev = new TransferDnaEvent { Donor = target, Recipient = implanter };
-                RaiseLocalEvent(target, ref ev);
+                    //Break so only one implant is drawn
+                    break;
+                }
 
-                //Break so only one implant is drawn
-                break;
+                if (component.CurrentMode == ImplanterToggleMode.Draw && !component.ImplantOnly && !permanentFound)
+                    ImplantMode(implanter, component);
+            }
+            else
+            {
+                var implant = implantContainer.ContainedEntities.FirstOrNull(entity => Prototype(entity) != null && component.DeimplantChosen == Prototype(entity)!);
+                if (implant != null && implantCompQuery.TryGetComponent(implant, out var implantComp))
+                {
+                    //Don't remove a permanent implant
+                    if (!_container.CanRemove(implant.Value, implantContainer))
+                    {
+                        DrawPermanentFailurePopup(implant.Value, target, user);
+                        permanentFound = implantComp.Permanent;
+
+                    }
+                    else
+                    {
+                        DrawImplantIntoImplanter(implanter, target, implant.Value, implantContainer, implanterContainer, implantComp);
+                        permanentFound = implantComp.Permanent;
+                    }
+
+                    if (component.CurrentMode == ImplanterToggleMode.Draw && !component.ImplantOnly && !permanentFound)
+                        ImplantMode(implanter, component);
+                }
+                else
+                {
+                    DrawCatastrophicFailure(implanter, component, user);
+                }
             }
-
-            if (component.CurrentMode == ImplanterToggleMode.Draw && !component.ImplantOnly && !permanentFound)
-                ImplantMode(implanter, component);
 
             Dirty(implanter, component);
+
+        }
+        else
+        {
+            DrawCatastrophicFailure(implanter, component, user);
         }
     }
 
+    private void DrawPermanentFailurePopup(EntityUid implant, EntityUid target, EntityUid user)
+    {
+        var implantName = Identity.Entity(implant, EntityManager);
+        var targetName = Identity.Entity(target, EntityManager);
+        var failedPermanentMessage = Loc.GetString("implanter-draw-failed-permanent",
+            ("implant", implantName), ("target", targetName));
+        _popup.PopupEntity(failedPermanentMessage, target, user);
```

Comment:
Is this predicted? It's using the non-predicted popup.

Reply:
Not predicted clientside.